npk/src/npk.rs

```diff
@@ -152,15 +154,15 @@ pub struct Npk<R: io::Read + io::Seek> {
 }
 
 impl<R: io::Read + io::Seek> Npk<R> {
-    pub fn new(npk: R) -> Result<Self, Error> {
+    pub async fn new(npk: R) -> Result<Self, Error> {
         let zip = zip::ZipArchive::new(npk).map_err(|e| Error::Zip {
```

Comment:
Probably we cannot construct a `ArchiveReader` from a `tokio::io::Read/Seek`.

Reply:
I think this is true. The `Npk` struct is a wrapper around a `ZipArchive` which does not support async. Most of our IO heavy operations go through the `zip` crate and are therefore synchronous. Slapping an async label on our struct will not change that until we find an async zip implementation.